src/utils/resolveProp.ts

```diff
@@ -1,21 +1,17 @@
-import { useDocument } from "@yext/pages/util";
-
-export const resolveProp = <T>(entityField: string): T => {
-  const document = useDocument() as any;
-
+export const resolveProp = <T>(entityField: string, document: any): T => {
   // check for the entity field in the document
   const steps: string[] = entityField.split(".");
   let missedStep = false;
   let current = document;
   for (let i = 0; i < steps.length; i++) {
-    if (current?.[steps[i]]) {
+    if (current?.[steps[i]] !== undefined) {
```

Comment:
undefined check is unnecessary

Reply:
"" is falsey, so this will not pass if we want to allow empty strings.